File: du-backend/src/main/java/com/dwin/du/entity/person/Person.java
package com.dwin.du.entity.person;
import com.dwin.du.entity.user.User;
import jakarta.persistence.*;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "person")
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "surname")
    private String surname;

    @Column(name = "receipts_count")
    private int receiptsCount;

    @Column(name = "total_amount")
    private Double totalAmount;

    @Column(name = "unpaid_amount")
    private Double unpaidAmount;
}

File: du-front/DivvyUp.Shared/Model/Person.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DivvyUp_Shared.Model
{
    public class Person
    {
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("User")]
        [Column("user_id")]
        public int UserId { get; set; }
        public User User { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; }

        [Column("surname")]
        public string Surname { get; set; }

        [Column("receipts_count")]
        public int ReceiptsCount { get; set; }

        [Column("products_count")]
        public int ProductsCount { get; set; }

        [Column("total_amount")]
        public decimal TotalAmount { get; set; }

        [Column("unpaid_amount")]
        public decimal UnpaidAmount { get; set; }

        [Column("loan_balance")]
        public decimal LoanBalance { get; set; }

        [Column("user_account")]
        public bool UserAccount { get; set; }
    }
}
